### ISSUE TITLE
[feathers-dove] using the createClient function

### ISSUE THREAD

**Issue by juni0r** [feathersjs/feathers#2713]
A newly generated app will have a `client.ts` file:

```javascript
import { feathers } from '@feathersjs/feathers'
import type { PostsData, PostsResult, PostsQuery } from './services/posts/posts.schema'
import type { Service, TransportConnection, Params } from '@feathersjs/feathers'

export interface ServiceTypes {
  posts: Service<PostsData, PostsResult, Params<PostsQuery>>
  // A mapping of client side services
}

export const createClient = <Configuration = any>(connection: TransportConnection<ServiceTypes>) => {
  const client = feathers<ServiceTypes, Configuration>()

  client.configure(connection)

  return client
}

```

I'm completely at loss on how to use `createClient` to obtain a REST or SocketIO client. The documentation on the client contains examples like

```javascript
import io from 'socket.io-client'
import { feathers } from '@feathersjs/feathers'
import socketio from '@feathersjs/socketio-client'

const socket = io('http://api.my-feathers-server.com')
const client = feathers()

client.configure(socketio(socket))
```

But trying to use this with `createClient(socketio(socket))` won't work since the methos expects a `TransportConnection<ServiceTypes>`. I don't even understand what this type represents (the type definition is not very enlightening).

Some examples would be greatly appreciated.

**Comment by daffl** [comment]
A working example can be found at https://github.com/feathersjs/feathers-chat/blob/dove/react-chat/src/feathers.ts - I will also add it to the CLI guide at https://dove.feathersjs.com/guides/cli/index.html shortly. Note that there was a bug on how the client is generated which will be fixed in the next prerelease (`5.0.0-pre.28`)  via https://github.com/feathersjs/feathers/pull/2701

**Comment by juni0r** [comment]
Ok, so I tried 

```javascript
import io from 'socket.io-client';
import socketio from '@feathersjs/socketio-client';
import authentication from '@feathersjs/authentication-client';
import { createClient } from './client';

const connection = socketio(io('http://localhost:3030'));
const client = createClient(connection);

client.configure(authentication() as any);
```

which results in
![image](https://user-images.githubusercontent.com/11867/182649107-651cef7d-b66d-4c5e-aa7d-af0def23c58c.png)

**Comment by daffl** [comment]
I just published v5.0.0-pre.28 with a bunch of fixes around this. Would you mind installing it and trying this with a newly generated app?

**Comment by juni0r** [comment]
My bad: I forgot to use `@pre` when I installed `@feathersjs/authentication-client`.

I now installed all packages from pre.28 and it works.

Thank you very much!
